File: Java/LeetCode/src/P3254AndP3255/Test02.java
package P3254AndP3255;

/*
* 3254. 长度为 K 的子数组的能量值 I
*https://leetcode.cn/problems/find-the-power-of-k-size-subarrays-i/?envType=daily-question&envId=2024-11-06
* 优化代码（暴力）
* */

import java.util.Arrays;

public class Test02 {
    public static int[] resultsArray(int[] nums, int k) {
        int[] ans = new  int[nums.length-k+1];
        Arrays.fill(ans,-1);
        for (int i = 0; i+k <= nums.length; i++) {

            boolean flag = true;
            for (int j = 1; j < k && flag; j++) {
                flag = (nums[j+i] == nums[i]+j);
            }

            if(flag){
                ans[i] = nums[i+k-1];
            }
        }
        return  ans;
    }
    public static void main(String[] args) {
        System.out.println(Arrays.toString(resultsArray(new int[]{1,2,3,4,3,2,5},3))); // [3,4,-1,-1,-1]
        System.out.println(Arrays.toString(resultsArray(new int[]{2,2,2,2,2},4))); // [-1,-1]
        System.out.println(Arrays.toString(resultsArray(new int[]{3,2,3,2,3,2},2))); // [-1,3,-1,3,-1]
    }
}

File: Python/LeetCode/P3233/Test02.py
"""
3233. 统计不是特殊数字的数字数量
https://leetcode.cn/problems/find-the-count-of-numbers-which-are-not-special/description/?envType=daily-question&envId=2024-11-22
只有质数的平方 p**2才是特殊数字
预处理 31622 内的质数
"""



class Solution:
    def nonSpecialCount(self, l: int, r: int) -> int:
        ans = 0
        if l == 1:
            ans+=1
            l = 2
        for i in range(l,r+1):
            g = i**0.5
            if g == int(g):
                for j in range(2,int(g)):
                    if  i%j==0:
                        ans+=1
                        break
                continue
            ans+=1
        return  ans

if __name__ == '__main__':
    print(Solution().nonSpecialCount(1, 2))  # 2
    print(Solution().nonSpecialCount(5,7)) #3
    print(Solution().nonSpecialCount(4,16))#11